sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -139,9 +147,14 @@ export class TableClient {
       }
     };
 
-    const pipeline = createPipelineFromOptions(internalPipelineOptions, credential);
+    let pipeline = createPipelineFromOptions(internalPipelineOptions, credential);
+
+    if (Array.isArray(clientOptions.requestPolicyFactories)) {
```

Comment:
I don't get this piece of code, why have your constructor take a method that modifies the requestPolicyFactories if you're only going to do something with the array form? You could just declare `clientOptions.requestPolicyFactories` to be an array and be done with it.

Also, since all `createPipelineFromOptions` does is return this shape:

```ts
return {
    httpClient: pipelineOptions.httpClient,
    requestPolicyFactories
  };
```

You don't even need to call `createPipelineFromOptions` if you get passed `requestPolicyFactories`, it just does a bunch of wasted work.

Reply:
Makes sense I'm taking your suggestion of getting taking InnerBatchRequest in TableServiceClientOptions. I'll check for that and conditionally call createPipelineFromOptions if it is not passed